update docs with docker toolbox & machine details

bbodenmiller opened the pull request:
Cleanup docs to reference Docker Toolbox and Machine where appropriate rather than depreciated `boot2docker` CLI tool.

yosifkit commented on README.md:
````diff
@@ -129,27 +67,12 @@ The following example will enable core dumps inside containers, but you can
 
 ```console
-boot2docker ssh -t sudo vi /var/lib/boot2docker/profile
+docker-machine default ssh -t sudo vi /var/lib/boot2docker/profile
````
Swap order of `ssh` and `default`.

yosifkit commented on README.md:
```diff
@@ -129,27 +67,12 @@ The following example will enable core dumps inside containers, but you can
 # Add something like:
 #     EXTRA_ARGS="--default-ulimit core=-1"
-boot2docker restart
+docker-machine
```
Probably still wanted restart here?  `docker-machine restart default`

yosifkit commented on README.md:
```diff
@@ -167,7 +90,7 @@ $ docker run -v /data --name my-data busybox true
 $ docker run --rm -v /usr/local/bin/docker:/docker -v /var/run/docker.sock:/docker.sock svendowideit/samba my-data
 $ # then find out the IP address of your Boot2Docker host
-$ boot2docker ip
+$ docker-machine ip
```
This still needs `default`, like all the docker-machine commands :cry:

yosifkit commented on README.md:
```diff
@@ -242,17 +165,15 @@ As of Docker version 1.3.1, if your registry doesn't support HTTPS, you must add
 
 So sometimes if you are behind a VPN, you'll get an `i/o timeout` error.
-The current work around is to forward the port in the boot2docker-vm.
+The current work around is to forward the port in the `boot2docker-vm`.
```
`boot2docker-vm` might get confused as a command.

yosifkit commented on README.md:
```diff
@@ -301,15 +214,15 @@ pass: tcuser
 will be automounted and used to persist your docker data in `/var/lib/docker`
 and `/var/lib/boot2docker`.  This virtual disk will be removed when you run
-`boot2docker delete`.  It will also persist the SSH keys of the machine.
+`docker-machine delete`.  It will also persist the SSH keys of the machine.
```
`default` again :wink:

yosifkit commented:
Thanks for taking this on!  I left a few comments, if you have some more time.  Thanks again.

bbodenmiller commented:
Addressed your comments. I believe technically `default` can be dropped from the `docker-machine` commands but it still demonstrates how you can target certain machines.

yosifkit commented:
LGTM, cc @tianon

tianon commented:
Very nice, thanks! :+1:

LGTM
